program-analysis/echidna/configuration.md

```diff
@@ -0,0 +1,382 @@
+
+## `allContracts`
+
+| Type | Default | Available in                    | CLI equivalent    |
+| ---- | ------- | ------------------------------- | ----------------- |
+| Bool | `false` | 2.1.0+ (previously `multi-abi`) | `--all-contracts` |
+
+Makes Echidna fuzz the provided test contracts and any other deployed contract whose ABI is known at runtime.
+
+## `filterBlacklist`
+
+| Type | Default | Available in |
+| ---- | ------- | ------------ |
+| Bool | `true`  | \*           |
+
+Allows Echidna to avoid calling (when set to true) or only call (when set to
+false) a set of functions. The function allowlist or denylist should be provided
+in `filterFunctions`.
+
+## `filterFunctions`
+
+| Type     | Default | Available in |
+| -------- | ------- | ------------ |
+| [String] | `[]`    | \*           |
+
+Configures the function allowlist or denylist from `filterBlacklist`. The list
+should contain strings in the format of
+`"Contract.functionName(uint256,uint256)"` following the signature convention.
+
+## `allowFFI`
+
+| Type | Default | Available in |
+| ---- | ------- | ------------ |
+| Bool | `false` | 2.1.0+       |
+
+Allows the use of the HEVM `ffi` cheatcode.
+
+## `rpcUrl`
+
+| Type   | Default | Available in | CLI equivalent  |
+| ------ | ------- | ------------ | --------------- |
+| String | `null`  | 2.2.0+       | `--rpc-url URL` |
+
+URL to fetch contracts over RPC.
+
+## `rpcBlock`
+
+| Type   | Default | Available in | CLI equivalent  |
+| ------ | ------- | ------------ | --------------- |
+| String | `null`  | 2.2.0+       | `--rpc-block N` |
+
+Block number to use when fetching over RPC.
+
+## `coverageFormats`
+
+| Type     | Default                 | Available in |
+| -------- | ----------------------- | ------------ |
+| [String] | `["txt","html","lcov"]` | 2.2.0+       |
+
+List of file formats to save coverage reports in; default is all possible
+formats.
+
+## `workers`
+
+| Type | Default | Available in | CLI equivalent |
+| ---- | ------- | ------------ | -------------- |
+| Int  | `1`     | 2.2.0+       | `--workers`    |
+
+Number of workers.
+
+## `server`
+
+| Type | Default | Available in | CLI equivalent  |
+| ---- | ------- | ------------ | --------------- |
+| Int  | `null`  | 2.2.2+       | `--server PORT` |
+
+Run events server on the given port.
+
+## `symExec`
+
+| Type | Default | Available in |
+| ---- | ------- | ------------ |
+| Bool | `false` | 2.2.4+       |
+
+Whether to add an additional symbolic execution worker.
+
+## `symExecConcolic`
+
+| Type | Default | Available in |
+| ---- | ------- | ------------ |
+| Bool | `true`  | 2.2.4+       |
+
+Whether symbolic execution will be concolic (vs full symbolic execution). Only
+relevant if `symExec` is true.
+
+## `symExecNSolvers`
+
+| Type | Default | Available in |
+| ---- | ------- | ------------ |
+| Int  | `1`     | 2.2.4+       |
+
+Number of SMT solvers used in symbolic execution. Only relevant if `symExec` is
```

Comment:
Clarify that there is only a single symexec worker, but we can have multiple threads to solve queries.